File: Extra - Estudos/GithubNivelamento/ex2/main.c
#include <stdio.h>
#include <stdbool.h>

bool binarySearch(int array[], int size, int x ){
    bool found = false;
    int dir = size - 1;
    int esq = 0;

    while(esq <= dir){
        int mid = (esq + dir)/2;
        if(x == array[mid]){
            return true;
        } else if (x > array[mid]){
            esq = mid + 1;
        } else {
            dir = mid - 1;
        }
    }
    return false;
}

int main(){
    int array[100], size, num;

    printf("Insira o tamanho da array: ");
    scanf("%d", &size);
    printf("Insira o número a ser encontrado: ");
    scanf("%d", &num);

    for(int i = 0; i < size ; i++){
        scanf("%d", &array[i]);
    }

    bool resul = binarySearch(array, size, num);

    if(resul == true){
        printf("Numero encontrado.\n");
    } else {
        printf("Numero nao encontrado.\n");
    }

    return 0;
}

File: Extra - Estudos/GithubNivelamento/ex2/Main.java
package ex2;

import java.util.*;

public class Main {

    public static boolean FindNumber(int dir, int[] arr, int x){
        int esq = 0;
        
        while(esq <= dir){
            int mid = dir/2;
            if(x == arr[mid]){
                return true;
            }else if(x < arr[mid]){
                dir = mid - 1;
            } else {
                esq = mid + 1;
            }
        }
        return false;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int size = sc.nextInt();
        int num = sc.nextInt();
        int[] arr = new int[size];  


        for(int i = 0 ; i < size ; i++){
            arr[i] = sc.nextInt();
        }

        boolean resp = FindNumber(size, arr, num);

        if (resp == true) {
            System.out.println("sim");
        } else {
            System.out.println("nao");
        }

    }
    
}
